### PR TITLE
Optional MCTruth handling for FDD digitization

### PR DESCRIPTION
**Pull request by sawenzel**

### CODE REVIEW

File: Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h [hunk 1]
```diff
@@ -27,17 +27,26 @@ namespace fdd
...
+                                  BranchDefinition<std::vector<o2::fdd::ChannelData>>{InputSpec{"digitChinput", "FDD", "DIGITSCH"}, "FDDDigitCh"},
+                                  BranchDefinition<o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>>{InputSpec{"labelinput", "FDD", "DIGITLBL"}, "FDDDigitLabels"})();
+  } else {
+    return MakeRootTreeWriterSpec("FDDDigitWriter",
+                                  "fdddigits.root",
+                                  "o2sim",
+                                  1,
+                                  BranchDefinition<std::vector<o2::fdd::Digit>>{InputSpec{"digitBCinput", "FDD", "DIGITSBC"}, "FDDDigit"},
+                                  BranchDefinition<std::vector<o2::fdd::ChannelData>>{InputSpec{"digitChinput", "FDD", "DIGITSCH"}, "FDDDigitCh"})();
+  }
```
Comment (sawenzel) [inline Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h:49]: @matthiasrichter : I wondering if there is a shorter way here to configure the list of BranchDefinitions based on an external parameter (essentially to influence the list of variadic arguments)?

File: Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h [hunk 2]
```diff
@@ -27,17 +27,26 @@ namespace fdd
...
-                                BranchDefinition<std::vector<o2::fdd::ChannelData>>{InputSpec{"digitChinput", "FDD", "DIGITSCH"}, "FDDDigitCh"},
-                                BranchDefinition<o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>>{InputSpec{"labelinput", "FDD", "DIGITLBL"}, "FDDDigitLabels"})();
+  if (mctruth) {
+    return MakeRootTreeWriterSpec("FDDDigitWriter",
+                                  "fdddigits.root",
+                                  "o2sim",
+                                  1,
+                                  BranchDefinition<std::vector<o2::fdd::Digit>>{InputSpec{"digitBCinput", "FDD", "DIGITSBC"}, "FDDDigit"},
+                                  BranchDefinition<std::vector<o2::fdd::ChannelData>>{InputSpec{"digitChinput", "FDD", "DIGITSCH"}, "FDDDigitCh"},
+                                  BranchDefinition<o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>>{InputSpec{"labelinput", "FDD", "DIGITLBL"}, "FDDDigitLabels"})();
```
Comment (matthiasrichter) [inline Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h:41]: There is a new feature to disable a defined branch, this make the code simpler as the compile time signature can be the same and branches are disabled at runtime by setting number of branches to '0'
```suggestion
                                  BranchDefinition<o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>>{InputSpec{"labelinput", "FDD", "DIGITLBL"}, "FDDDigitLabels", (mctruth ? 1 : 0)})();
```

**Review by matthiasrichter — COMMENTED** [review]
Very good. The code can be simplified a bit using a feature of 4cc22279. I have started to simplify other places, but its not yet in a PR.

File: Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h (hunk 2 above)
Comment (matthiasrichter) [reply Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h:41]: think it should work like that, let me know.

File: Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h [hunk 3]
```diff
@@ -27,17 +27,26 @@ namespace fdd
...
+  } else {
```
Comment (matthiasrichter) [inline Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h:42]: then the two code branches reduce to only one with the runtime enable/disable for the MC branch definition.

File: Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h (hunk 1 above)
Comment (matthiasrichter) [reply Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h:49]: I assume your question goes in the direction I have answered above?

Comment (matthiasrichter) [reply Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h:49]: below :-)

File: Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h (hunk 2 above)
Comment (sawenzel) [reply Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h:41]: Nice. I'll try.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**